src/storage/btree.c

```diff
@@ -30599,8 +30603,9 @@ btree_key_delete_remove_object (THREAD_ENTRY * thread_p, BTID_INT * btid_int, DB
 		  logtb_find_current_mvccid (thread_p)))
 	  /* Cannot check if class OID is NULL. Get it in debug mode. */
 	  || OID_ISNULL (BTREE_DELETE_CLASS_OID (delete_helper))
-	  || lock_has_lock_on_object (BTREE_DELETE_OID (delete_helper), BTREE_DELETE_CLASS_OID (delete_helper),
-				      logtb_get_current_tran_index (), X_LOCK) > 0);
+	  || (LOG_ISTRAN_ABORTED (LOG_FIND_CURRENT_TDES (thread_p))
```

Comment:
this safe-guard is too loose. it allows you to have no locks on abort.
the best safe-guards:
  - should allow BU_LOCK on delete&abort
  - should require instance locking otherwise

Maybe it is better to have two different functions btree_check_locking_for_insert_unique and btree_check_locking_for_delete_unique.